server/game/cards/03-WotN/GreenDreams.js

```diff
@@ -10,16 +10,18 @@ class GreenDreams extends DrawCard {
                 this.topCard = this.controller.drawDeck[0];
                 this.game.addMessage('{0} uses {1} to look at the top card of their deck', this.controller, this);
 
-                this.game.promptWithMenu(this.controller, this, {
-                    activePrompt: {
-                        menuTitle: 'Put ' + this.topCard.name + ' on bottom of your deck?',
-                        buttons: [
-                            { text: 'Yes', method: 'placeOnBottom', arg: 'yes', card: this.topCard },
-                            { text: 'No', method: 'placeOnBottom', arg: 'no', card: this.topCard }
-                        ]
-                    },
-                    source: this
-                });
+                if(this.topCard) {
```

Comment:
1. If there isn't a top card, shouldn't we not allow the ability to trigger at all? The better fix would be to check it in the triggering condition.
2. If there is no cards in the draw deck, you lose immediately.

Reply:
good catch, now that you mention the lose condition, i wonder how this happened in the first place. I remember seeing this in a stacktrace in the server log as the reason for the gamenode crashing.
Probably pulling it up to the triggering condition is a better idea